### DISCUSSION TITLE
2018 Mini, 2020 Mini, 2021 MBP and 2022 Mac Studio HDMI output + hardware DDC control?

### DISCUSSION THREAD

**Discussion by waydabber** [MonitorControl/MonitorControl#750] [Q&A] [answered]
Why is it that HDMI connected displays cannot be controlled via DDC on the following macs?

- Intel Mac Mini 2018
- M1 Mac mini 2020
- M1 Pro/Max MacBook Pro 2021
- Mac Studio

**Comment by waydabber** [comment, marked as answer]
_UPDATE - 3rd of March, 2023:_

The following apps have support for control via the built-in HDMI port as of now:

- [BetterDisplay](https://betterdisplay.pro) - Built in HDMI ports of all **Apple Silicon Macs** (all versions of M1 and M2 chips) and also the **2018 Intel mini**.
- [Lunar](https://lunar.fyi) - Built-in HDMI port of all **Apple Silicon Macs** (2018 Intel mini is not supported as of now)

MonitorControl is currently not updated, contributors are welcome to add a Pull Request with the required changes!

_Note: all DDC feature is free with BetterDisplay, no limitations. DDC features are free for up to 100 adjustments/day with Lunar._

---

### Explanation

The Macs listed in the question produce DisplayPort signal which is converted to HDMI via a special chip, the [MCDP2900 DisplayPort 1.4a-to-HDMI 2.0b Converter](https://www.kinet-ic.com/mcdp2900/).

This chip is able to receive Displayport 1.4 signal and convert it to HDMI 2.0. Apple uses the very same design for all of these Macs (hence the HDMI 2.0 limitation even on the newest Macs).

~~It seems, even though this chip is capable of performing I2C communication (which is required both for DDC control and retrieving display information - EDID), the communication is cut off at firmware level once the initial negotiation and download of the EDID is completed (alternatively, there might be a yet undiscovered method or service endpoint that would allow communication).~~

~~Since this seems to be a design decision on Apple's part and reprogramming this chip (if it is even possible) would require a properly signed firmware update from Apple, there is little chance that DDC control via HDMI will be resolved on newer Macs.~~

### What can I do to change this situation?

~~You can contact Apple support and explain the situation and ask for the issue to be escalated to the engineers. You can also use the feedback assistant to add your voice.~~

Apple does not officially support or adhere to the DDC standard.

**Reply by kevinmpowell** [reply]
To clarify, if I use a USB-C to DisplayPort adapter _instead_ of a USB-C to HDMI adapter will keyboard volume controls work again?

**Reply by waydabber** [reply]
The problem is with the HDMI port of the Mac only. Most USB-C to HDMI adapters should work as well. M1 Macs can control displays via USB-C mode (by using DisplayPort Alt). Typical USB-C to HDMI dongles usually use DisplayPort Alt Mode internally to communicate with the computer. The question is if the converter chip inside the dongle (which converts DisplayPort data to HDMI) do relay bidirectional DDC i2c communication properly or not. Some (most) USB-C to HDMI dongles do, however some don't.

**Reply by SamShylla** [reply]
Hi, big fan here! 

Need help!

Wanted to know the reason why the USB-C to display port cable for a MacBook M2 works only for a while and then stops working. Is it necessary to always use an adaptor instead?

Details:
USB-C to display port 6ft-Cable Matters
MacBook Air M2
External monitor- Dell U2718Q

**Reply by waydabber** [reply]
No reason, it should work just fine. Something else is wrong probably.

**Reply by SamShylla** [reply]
Just curios though. What would be the best option?

And, thanks a ton!

**Reply by waydabber** [reply]
A direct USB-C to DisplayPort cable is among the best option as it normally does not need any complicated conversion, USB-C natively carries DisplayPort data (DisplayPort Alt mode).

**Reply by SamShylla** [reply]
We exactly did that and it's so confusing now. A brand new MacBook Air with brand new cables worked fine earlier and suddenly stopped! 
A software issue maybe?

Don't mind me for being direct. But, do you take private consultations? My Boss is willing to pay.
